proto/viam/provisioning/provisioning.proto

```diff
@@ -0,0 +1,90 @@
+syntax = "proto3";
+
+package viam.provisioning.v1;
+
+option go_package = "go.viam.com/api/provisioning/v1";
+
+// ProvisionService is used by viam-agent to retrieve device and robot info.
+service ProvisionService {
+  // GetDeviceStatus is for retrieving the status of the device including networking.
+  rpc GetDeviceStatus(GetDeviceStatusRequest) returns (GetDeviceStatusResponse);
+
+  // SetWifiCredentials is to set the wifi credentials.
+  rpc SetWifiCredentials(SetWifiCredentialsRequest) returns (SetWifiCredentialsResponse);
+
+  // SetRobotCredentials is to set the robot credentials.
+  rpc SetRobotCredentials(SetRobotCredentialsRequest) returns (SetRobotCredentialsResponse); 
+
+  // GetVisibleWifiNetworks is to retrieve the list of wifi networks that are visible to the device.
+  rpc GetVisibleWifiNetworks(GetVisibleWifiNetworksRequest) returns (GetVisibleWifiNetworksResponse); 
+}
+
+message GetDeviceStatusRequest {
+}
+
+message GetDeviceStatusResponse {
+  ProviosionInfo provision_info = 1;
+  ConnectionAttempt latest_connection_attempt = 2;
+}
+
+message SetWifiCredentialsRequest {
+  string ssid = 1;
+  string psk = 2;
+}
+
+message SetWifiCredentialsResponse {
+  ConnectionAttempt connection_attempt = 1; 
+}
+
+message SetRobotCredentialsRequest {
+}
+
+message SetRobotCredentialsResponse {
+  CloudConfig cloud = 1;
+}
+
+message GetVisibleWifiNetworksRequest {
+}
+
+message GetVisibleWifiNetworksResponse {
+  repeated ConnectionInfo wifis = 1;
+}
+
+message ProviosionInfo { 
+  string fragment_id  = 1;
+  string model = 2;
+  string manufacturer = 3;
+}
+
+message ProvisionStatus {
```

Comment:
I'm not using this yet... just an idea and i don't like the name. If we wanted to add context to device status about the subsystems, maybe we could do something like this.

Reply:
Let's not add it until we need it.